frontend/src/metabase/common/hooks/use-setting/use-setting.ts

```diff
@@ -40,3 +48,36 @@ export const useUserSetting = <T extends keyof UserSettings>(
   );
   return [currentValue, shouldDebounce ? debouncedSetter : setter];
 };
+
+export const useMergeSetting = <Key extends SettingKey>(
+  displaySetting: SettingElement<Key>,
+): SettingElement<Key> => {
+  const settingDefinitions = useSelector(getAdminSettingDefinitions);
+  const apiSetting = settingDefinitions.find(
+    (setting: SettingDefinition) => setting.key === displaySetting.key,
+  ) as SettingDefinition<Key> | undefined;
```

Comment:
TS was giving me ridiculous, incomprehensible errors about how I couldn't use the key from apiSetting to define a SettingDefinition, so I kinda had to do this. When I have more time I'll look at it again